### PR TITLE
PPML rename DataType and package refactor

### PR DESCRIPTION
**Pull request by Litchilitchy**

### CODE REVIEW

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/base/StorageHolder.scala [hunk 1]
```diff
@@ -16,45 +16,63 @@
...
+      logger.debug(s"Put TreeEval into client data map, current size: $clientDataSize")
+    } else {
+      throw new IllegalArgumentException("Data is empty, could not uploaded to server.")
     }
   }
-  def getTableStorage() = this.tableStorage
-  def getSplitStorage() = this.splitStorage
-  def getLeafStorage() = this.leafStorage
-  def getBranchStorage() = this.branchStorage
+  def getTableStorage() = this.tensorMapStorage
```
Comment (jason-dai) [inline scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/base/StorageHolder.scala:74]: getTensorMapStorage

File: scala/ppml/src/main/java/com/intel/analytics/bigdl/ppml/common/FLDataType.java [hunk 2]
```diff
@@ -19,6 +19,6 @@
 public enum FLDataType {
     TENSOR_MAP,
     TREE_SPLIT,
-    TREE_LEAF,
+    TREE_LEAVES,
```
Comment (jason-dai) [inline scala/ppml/src/main/java/com/intel/analytics/bigdl/ppml/common/FLDataType.java:22]: Why TREE_LEAVES instead of TREE_LEAF?

Comment (Litchilitchy) [reply scala/ppml/src/main/java/com/intel/analytics/bigdl/ppml/common/FLDataType.java:22]: This actually store an array of leaves. However this is just a naming and we'd better fix all the names the same.

Comment (jason-dai) [reply scala/ppml/src/main/java/com/intel/analytics/bigdl/ppml/common/FLDataType.java:22]: treeLeavesStorage sounds very awkward

Comment (Litchilitchy) [reply scala/ppml/src/main/java/com/intel/analytics/bigdl/ppml/common/FLDataType.java:22]: How about change the type to `TREE_LEAF` and `getTreeLeafStorage()` and remain the proto `TreeLeaves'

Comment (jason-dai) [reply scala/ppml/src/main/java/com/intel/analytics/bigdl/ppml/common/FLDataType.java:22]: I think we should keep them consistent - either LEAF or LEAVES for all

Comment (Litchilitchy) [reply scala/ppml/src/main/java/com/intel/analytics/bigdl/ppml/common/FLDataType.java:22]: So change to LEAF for all?

**Pull request merged**